File: aire-core/src/main/java/io/zephyr/aire/elements/AireCard.java
package io.zephyr.aire.elements;

import com.vaadin.flow.component.Component;
import com.vaadin.flow.component.Composite;
import com.vaadin.flow.component.dependency.CssImport;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.html.Section;

@CssImport("./styles/aire/components/aire-card.css")
public class AireCard extends Composite<Div> {

  private Div icon;
  private Section content;
  private AireHeader header;
  private AireFooter footer;

  public AireCard() {}

  public void setContent(Component content) {}
}

File: aire-core/src/main/frontend/components/aire/aire-card.ts
import {
  TemplateResult,
  LitElement,
  html,
  customElement
} from 'lit-element';

/// TS_IGNORE
@customElement('aire-card')
class AireCard extends LitElement {


  render() : TemplateResult {
    return html`
        <slot name="header"></slot>
        <slot name="content"></slot>
        <slot name="footer"></slot>
    `;

  }

}

